Quaver.Shared/Screens/Gameplay/Rulesets/Keys/Playfield/Hits/HitManager.cs

```diff
@@ -0,0 +1,143 @@
+using System.Collections.Generic;
+using System.Diagnostics;
+using Quaver.API.Maps.Processors.Scoring.Data;
+using Quaver.Shared.Screens.Gameplay.Rulesets.Input;
+using Quaver.Shared.Screens.Gameplay.Rulesets.Keys.HitObjects;
+
+namespace Quaver.Shared.Screens.Gameplay.Rulesets.Keys.Playfield.Hits
+{
+    public class HitManager
```

Comment:
Instead of creating an entirely new manager and container for this, why not have the hits be children of the hitobjects themselves? Creating a new pool for this seems a bit redundant when the hitobjects already have that system in place and the hits act that way anyway. It should be a lot simpler this way as well.

Reply:
That's what my initial implementation was, but the main issue was that there's no good correspondence between hit objects and their hit stats, so I ended up doing a linear search over all hit stats for every hit object, twice (press and release). It was actually like 2x more code (because you need to handle both presses and releases for each hit object).